builtin.c

```diff
@@ -273,33 +272,26 @@ XS(XS_builtin_trim)
         }
     }
 
-    dest = TARG;
+    SV_CHECK_THINKFIRST(TARG);
+    SvUPGRADE(TARG, SVt_PV);
+    SvGROW(TARG, len + 1);
```

Comment:
Is grow required here? Fairly sure that trim can't make an SV bigger, only smaller, so surely it must already have enough storage for the new size.

Reply:
TARG and source are not the same SV.

If this were a pp func and implemented TARGLEX, then we'd retain the branch I removed and do some special handling for that.

But it's XS, so `TARG` shouldn't be `source` (though source might be the TARG from another entersub calling trim, or from the same entersub in recursion, in either case it's not the TARG here.

Reply:
Ohright of course, this isn't an in-place trim, it's a substr copy.